.github/workflows/ci.yml

```diff
@@ -0,0 +1,24 @@
+name: CI
+on:
+  pull_request:
+    branches: [ "main" ]
+    
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.22.2'
```

Comment:
The go version that Scribe-Server uses is `1.20` (as in the [`go.mod`](https://github.com/scribe-org/Scribe-Server/blob/3e336ef24dec7eaf6c1d2a6d0dd7088f4b815901/go.mod#L3) file). This is due to the version that was available in Toolforge, where Scribe-Server is hosted.
We should probably test against the same version that Scribe-Server is on :thinking: That got me thinking though.. could we get this action to read the version off the `go.mod` file in case we change the version? I checked the [docs](https://github.com/actions/setup-go/?tab=readme-ov-file#getting-go-version-from-the-gomod-file) and lo and behold, it looks like we can! :rocket: Doing this would be great